Realm/Tests/QueryTests.m

```diff
@@ -1929,4 +1922,269 @@ - (void)testSortingColumnsWithNull {
     [realm cancelWriteTransaction];
 }
 
+struct NullTestData {
+    __unsafe_unretained NSString *propertyName;
+    __unsafe_unretained NSString *nonMatchingStr;
+    __unsafe_unretained NSString *matchingStr;
+    __unsafe_unretained id nonMatchingValue;
+    __unsafe_unretained id matchingValue;
+    bool orderable;
+    bool substringOperations;
+};
+
+- (void)testEqualityOnNull {
+    RLMRealm *realm = [RLMRealm defaultRealm];
+
+    // nil on LHS is currently not supported by core
+    XCTAssertThrows([AllOptionalTypes objectsWhere:@"nil = boolObj"]);
+
+    // These need to be stored in variables because the struct does not retain them
+    NSData *matchingData = [@"" dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *notMatchingData = [@"a" dataUsingEncoding:NSUTF8StringEncoding];
+    NSDate *matchingDate = [NSDate dateWithTimeIntervalSince1970:1];
+    NSDate *notMatchingDate = [NSDate dateWithTimeIntervalSince1970:2];
+
+    struct NullTestData data[] = {
+        {@"boolObj", @"YES", @"NO", @YES, @NO},
+        {@"intObj", @"1", @"0", @1, @0, true},
+        {@"floatObj", @"1", @"0", @1, @0, true},
+        {@"doubleObj", @"1", @"0", @1, @0, true},
+        {@"string", @"'a'", @"''", @"a", @"", false, true},
+        {@"data", nil, nil, notMatchingData, matchingData},
+        {@"date", nil, nil, notMatchingDate, matchingDate, true},
+    };
+
+    // Assert that the query "prop op value" gives expectedCount results when
+    // assembled via string formatting
+#define RLMAssertCountWithString(expectedCount, op, prop, value) \
+    do { \
```

Comment:
It makes the macro expand to the equivalent of a single statement that requires a trailing semicolon, which avoids a bunch of potential bugs from macro-expansion edge cases.